### ISSUE TITLE
Allow Resources to be saved as Drafts

### ISSUE THREAD

**Issue by cheapscate7** [aberonni/improvdb#18]
I wanted to add a resource but I didn't have time to do it all in one sitting so I had to copy stuff into a temporary location until I have the time to finish it. I thought it would be nice to have a Drafts feature for new Resources or cloned Resources.

### How it would work

- Admins would not be alerted to drafts and cannot view them in the app unless they have the Resource ID. 
- The author of a draft can access it from "My Proposed Resources", but under a different tab to submitted Resources.
- "Save as draft" would be a checkbox that is visible _only_ when the resource hasn't been submitted for review _nor_ when you are proposing a change. This should help keep the workflow simple
- Normal users who are not the author cannot view the resource at all
- To begin with at least, the ID of the resource cannot be changed once it is in the database, again just to keep things simple

### Why?

It might encourage users to contribute or submit more quality resources but mainly it is QOL improvement

**Comment by cheapscate7** [comment]
I have already been working on a PR for this issue FYI

**Comment by aberonni** [comment]
This is really cool, thanks! Let me know if you need any help with this ✌️

**Comment by cheapscate7** [comment]
@aberonni could I have the permissions to push branches?

**Comment by aberonni** [comment]
I'd rather not - I think we can stick to regular OSS PR flow for now. If this flow is new to you, this might be of help: https://dev.to/idurar/a-simple-guide-to-creating-a-pull-request-on-github-3daa

**Comment by aberonni** [comment]
You can't push directly to the origin on `aberonni/improvdb` because you are not a maintainer on this repository. That is by design - as in, I prefer not to give maintainer permissions unless strictly necessary. In this case, it is not strictly necessary because you can contribute to the project without obtaining these extra permissions. You can do this by contributing to your own fork, and then creating a PR here through that. This is the typical flow for contributing to projects that you are not a maintainer of, rather than asking for extra permissions.

So to recap, the ideal flow is:
- fork the repo (thus creating a `cheapscate7/improvdb` repository)
- push your new branch to your the newly created repo
- create a PR from your `cheapscate7/improvdb` fork towards the `aberonni/improvdb` repo (GitHub will automatically suggest this when you push your new branch to your new fork)

This flow is outlined in more detail in the article I linked above (including the extra steps of creating your own fork and pushing to that). Let me know if there is still confusion, I'm happy to help further if I can :)
